src/coverage.ts

```diff
@@ -0,0 +1,86 @@
+import { fork, execSync } from 'child_process';
+import path from 'path';
+import fs from 'fs';
+
+export function cleanUp(): void {
+  if (fs.existsSync('./contracts-backup/')) {
+    execSync('cp -rf ./contracts-backup/ ./contracts');
+  }
+  execSync('rm -rf ./contracts-backup/');
+  execSync('rm -rf ./build/contracts/');
+  execSync('rm -rf ./.coverage_artifacts/');
+  execSync('rm -rf ./.coverage_contracts/');
+}
+
+['SIGINT', 'SIGTERM', 'SIGQUIT'].forEach((signal) =>
+  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+  // @ts-ignore
+  process.on(signal, () => {
+    cleanUp();
+    process.exit();
+  }),
+);
+
+/* eslint-disable @typescript-eslint/no-var-requires */
+export async function runCoverage(skipFiles: string[], compileCommand: string, testCommand: string): Promise<void> {
+  const client = require('ganache-cli');
+  const CoverageAPI = require('solidity-coverage/api');
+  const utils = require('solidity-coverage/utils');
+
+  const api = new CoverageAPI({ client });
+
+  const config = {
+    workingDir: process.cwd(),
+    contractsDir: path.join(process.cwd(), 'contracts'),
+    logger: {
+      log: (msg: string): void => console.log(msg),
+    },
+  };
+
+  try {
+    const { tempContractsDir, tempArtifactsDir } = utils.getTempLocations(config);
+    utils.setupTempFolders(config, tempContractsDir, tempArtifactsDir);
+
+    const { targets } = utils.assembleFiles(config, skipFiles);
+    const instrumented = api.instrument(targets);
+    utils.save(instrumented, config.contractsDir, tempContractsDir);
+
+    // backup original contracts
+    execSync('cp -rf ./contracts/ ./contracts-backup');
+    execSync(`cp -rf ./.coverage_contracts/ ./contracts`);
```

Comment:
1. Is it possible to move the directories rather than copy them? I feel it has less of a chance of causing problems.
2. If possible I think we should use Node.js modules rather than bash commands, just to ensure as much compatibility as possible. Moving the directory will also better in that sense because it isn't a recursive operation.